premake5.lua

```diff
@@ -103,6 +110,16 @@ if not BUILD_SQLITE then
     SQLITE_LIB_DIR = GetParam("sqlite-lib-dir") or "/usr/local/lib"
 end
 
+if GetParam("build-miniaudio") then
+    BUILD_MINIAUDIO = true
+elseif GetParam("no-build-miniaudio") then
+    BUILD_MINIAUDIO = false
+end
+if not BUILD_miniaudio then
+    MINIAUDIO_INCLUDE_DIR = GetParam("miniaudio-include-dir") or "/usr/local/include"
```

Comment:
miniaudio being a single header library, everything in premake5.lua is not needed.
You may toss the miniaudio.h file to gframe directory directly. But remember to add some conditional checks in `gframe/premake5.lua`